### PR TITLE
Clean up FieldValueTests

### PR DESCRIPTION
**Pull request by wilhuff**
Reorganize to make things clearer:
  * Split out the tests that are really exercising the `UserDataConverter` into a `UserDataConverterTest`.
  * Move the Timestamp tests that are unrelated to `FieldValue` into `TimestampTest`

Also fix nullability annotations on `Timestamp`.

### CODE REVIEW

File: firebase-firestore/src/main/java/com/google/firebase/Timestamp.java [hunk 1]
```diff
@@ -37,7 +38,7 @@
 @PublicApi
 public final class Timestamp implements Comparable<Timestamp>, Parcelable {
 
-  @PublicApi
+  @PublicApi @NonNull
```
Comment (mikelehen) [inline firebase-firestore/src/main/java/com/google/firebase/Timestamp.java]: Newline?

Comment (wilhuff) [reply firebase-firestore/src/main/java/com/google/firebase/Timestamp.java]: I had a newline here but `googleJavaFormat` eated it.

Comment (mikelehen) [reply firebase-firestore/src/main/java/com/google/firebase/Timestamp.java]: How rude!

**Pull request merged**